### PR TITLE
Chore: Remove core-js polyfills

### PR DESCRIPTION
**Pull request by leeoniya**
for a long time i've seen `push()` in the my perf profiles and always wondered why we need to polyfill an Array method that has existed since the days of IE6. so i went down a rabbit hole to drop some polyfills which likely serve no purpose any more. this removed 800kb of JS (out of the full 22MB unminified production build).

along the way i had to fork the unmaintained `sql-formatter-plus` to drop its polyfill and lodash dependencies, and bring the codebase into the 21st century: https://github.com/grafana/sql-formatter-plus. all its tests still pass after the refactor but would be good for someone to test the sql auto-formatting in the sql code editor.

image

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -12904,20 +12893,20 @@ __metadata:
...
 
-"core-js@npm:3.33.0, core-js@npm:^3.6.0, core-js@npm:^3.8.3":
-  version: 3.33.0
-  resolution: "core-js@npm:3.33.0"
-  checksum: 940ad9c3bbc971d884c08da7cf4a3fe9b360011a35b9e93c35627c439419f0e852749a357bef5d072d5f8921d45b46bd886a3ebb6979628c292ecf787a50506f
-  languageName: node
-  linkType: hard
-
-"core-js@npm:^2.4.0, core-js@npm:^2.6.5":
+"core-js@npm:^2.4.0":
```
Comment (leeoniya) [inline yarn.lock:12896]: i guess now one step closer to removing core-js 2.x? :shrug:

**Comment by leeoniya** [comment]
in FlameGraph rendering it's a significant cost.

before:
image

after:
image

**Comment by leeoniya** [comment]
not sure why `test-a11y-frontend` tests are failing, but hopefully can be solved without adding back 800kb of runtime code 😅

**Review by ashharrison90 — COMMENTED** [review]
@leeoniya rather than forking `sql-formatter-plus`, can we instead just use `sql-formatter`? given https://github.com/sql-formatter-org/sql-formatter/pull/112, i think they've merged all the extra changes back in to `sql-formatter`, and that library is well supported

**Comment by leeoniya** [comment]
i considered it, but that lib is still monolithic (non-shakeable) and 10x larger :cry: 

image

**Comment by ashharrison90** [comment]
can we feed any updates back to the main repo then? 🤔 

imo the burden of maintaining the fork isn't worth 800kb of javascript savings 😬

**Comment by leeoniya** [comment]
wonder if the pa11y stuff would be fixed by https://github.com/pa11y/pa11y/pull/681. some big changes in there.

**Comment by leeoniya** [comment]
> can we feed any updates back to the main repo then? 🤔

it looks inactive, as does the author. i dont think this will happen any time soon.

> imo the burden of maintaining the fork isn't worth 800kb of javascript savings 😬

it hasn't actually been maintained for 4 years, and we dont need to "maintain it". all i did was drop its runtime dependencies. if it ever stops working for us, we will simply switch to something that works. i suspect that will be quite far into the future.

**Comment by ashharrison90** [comment]
sorry, i meant "can we feed updates back to the main `sql-formatter` repo and switch to that instead", not this fork that i agree is definitely dead 👍 

> it hasn't actually been maintained for 4 years, and we dont need to "maintain it". all i did was drop its runtime dependencies. if it ever stops working for us, we will simply switch to something that works. i suspect that will be quite far into the future.

hard disagree. we do need to maintain it, otherwise that code is just going to sit there and gradually rot. as you said we've been on a fork for the last 4 years and missed out on [around ~2400 commits worth of updates](https://github.com/sql-formatter-org/sql-formatter/compare/v3.0.0...master) from the main repo. who knows what performance updates or bug fixes that contains? 🤔

**Comment by leeoniya** [comment]
> Are we sure it's not shakeable? Unless I have the wrong package, [package.json has esm and exports](https://unpkg.com/browse/sql-formatter@14.0.0/package.json) which webpack should pick up and be able to treeshake no?

its api is one function (same as it was 4 years ago), AFAICT: https://github.com/kufii/sql-formatter-plus#usage

> sorry, i meant "can we feed updates back to the main sql-formatter repo and switch to that instead", not this fork that i agree is definitely dead 👍

that lib now has different, trickier deps now that will be harder to remove, and it is monolithic, so will not have the same ROI.

> hard disagree. we do need to maintain it, otherwise that code is just going to sit there and gradually rot. as you said we've been on a fork for the last 4 years and missed out on [around ~2400 commits worth of updates](https://github.com/sql-formatter-org/sql-formatter/compare/v3.0.0...master) from the main repo. who knows what performance updates or bug fixes that contains? 🤔

well, it has 10x the amount of code. it probably has more new bugs than bugfixes in it. and given that it's a sql formatter (with no deps after my changes), i doubt that any of them are security-related.

if you insist, i will revert the `sql-formatter-plus` changes. they actually have no effect on the savings here because we still end up shipping core-js 2.x because of the `rc-` dependencies, but i think that core-js import is scoped and not env-wide.

still, i think it's weird to reason that it's better to keep what we currently have than to make a material improvement to the situation that carries little risk :upside_down_face:.

**Comment by ashharrison90** [comment]
i'm not _insisting_... i'll accept it if i'm outvoted 🙏 but imo forking things like that ends up being more burden than it's worth 😅

**Comment by leeoniya** [comment]
> i'm not insisting... i'll accept it if i'm outvoted 🙏 but imo forking packages almost always ends up being more burden than it's worth 😅

let's bikeshed it in another PR, since it has no effect here anyways :stuck_out_tongue_closed_eyes:

**Review by ashharrison90 — APPROVED** [review]
lgtm 👍

File: packages/grafana-ui/package.json [hunk 2]
```diff
@@ -164,6 +163,7 @@
     "@types/tinycolor2": "1.4.3",
     "@types/uuid": "9.0.2",
     "common-tags": "1.8.2",
+    "core-js": "3.33.0",
```
Comment (joshhunt) [inline packages/grafana-ui/package.json:166]: do we still need this?

Comment (leeoniya) [reply packages/grafana-ui/package.json]: nope, good catch. also clean up another place.

Comment (leeoniya) [reply packages/grafana-ui/package.json]: ah, never mind. we need them in devdeps for the tests which do `import 'core-js/stable/structured-clone';`

**Pull request merged**